redisson/src/main/java/org/redisson/connection/SentinelConnectionManager.java

```diff
@@ -327,6 +357,8 @@ private void checkState(SentinelServersConfig cfg, Iterator<RedisClient> iterato
             if (lastException.get() != null) {
                 log.error("Can't update cluster state", lastException.get());
             }
+            // If all sentinels are dead, set sentinelFlag = 1
+            sentinelFlag = 1;
```

Comment:
Can we just drop all sentinels in list at that moment? so they be reinit at next performSentinelDNSCheck call?

```java
        sentinels.keySet().stream()
                .forEach(uri -> {
                    RedisClient sentinel = sentinels.remove(uri);
                    if (sentinel != null) {
                        disconnectNode(uri);
                        sentinel.shutdownAsync();
                        log.warn("sentinel: {} is down", uri);
                    }
                });
```